### ISSUE TITLE
Newly enabled tests in test_groupby fail.

### ISSUE THREAD

**Issue by joergdietrich** [dask/dask#3481]
The tests enabled by PR #3480 contain some that fail. These are

```
    pdf = pd.DataFrame({'x': [0, 1, 2, 3, 4, 6, 7, 8, 9, 10],
                        'y': list('abcbabbcda')})
    ddf = dd.from_pandas(pdf, 3)

    gp = pdf.groupby('y')['x']
    dp = ddf.groupby('y')['x']
    assert isinstance(dp.obj, dd.Series)
```

and

```
    gp = pdf.groupby(pdf.y)['x']
    dp = ddf.groupby(ddf.y)['x']
    assert isinstance(dp.obj, dd.Series)
```

**Comment by mrocklin** [comment]
Yeah, it looks like we just represent things differently internally.  I'm not sure this is a bug so much as just a different approach.  Perhaps this is why the test was semi-removed?

**Comment by joergdietrich** [comment]
`dp.obj` in both cases is

```
In [12]: dp.obj
Out[12]: 
Dask DataFrame Structure:
                   x       y
npartitions=3               
0              int64  object
4                ...     ...
8                ...     ...
9                ...     ...
Dask Name: from_pandas, 3 tasks
```

Why would you argue that a dataframe containing `x` and `y` is the correct representation when only the column `x` is selected? I'm not sure why a user would ever need to access .obj (I'm not even sure what it is supposed to do), but if this is supposed to be a user facing attribute

```
assert_eq(dp.obj, gp.obj)
```

should not fail.

I'm not vehemently in favor of fixing these tests. I'm just not sure that these aren't bugs.

On the semi-removal: I'm not sure that code was _ever_ run. In the current version already the initialization of the pandas DataFrame would have failed with a ValueError.

**Comment by mrocklin** [comment]
Sorry for the lack of response here @joergdietrich .  I don't know the answer to your question.  If you have a thought about how to fix things within the dask dataframe codebase I encourage you to implement it.
